server/src/main/java/org/opensearch/transport/InboundDecoder.java

```diff
@@ -185,7 +185,7 @@ private int headerBytesToRead(BytesReference reference) {
     }
 
     // exposed for use in tests
-    static Header readHeader(Version version, int networkMessageSize, BytesReference bytesReference) throws IOException {
+    public static Header readHeader(Version version, int networkMessageSize, BytesReference bytesReference) throws IOException {
```

Comment:
Kind of a theme here...I don't see where this change is needed. Am I missing something? I always scrutinize visibility changes when they increase visibility, so I'd prefer that these changes are only made when a corresponding change that requires them is also present.

Reply:
yes, these are needed in the next PR where flight based streaming transport is implemented in a plugin. I will move these changes to the other PR itself.